File: programmers/lv2/math/테이블 해시 함수.py
def solution(data, col, row_begin, row_end):
    answer, arr = 0, []
    data.sort(key=lambda x:x[0], reverse=True)
    data.sort(key=lambda x:x[col - 1])
    for i in range(row_begin - 1, row_end):
        num_sum = 0
        for num in data[i]:
            num_sum += num % (i + 1)
        arr.append(num_sum)
    for num in arr:
        answer = answer ^ num
    
    return answer

File: programmers/lv2/연습문제/테이블 해시 함수.js
function solution(data, col, row_begin, row_end) {
    var answer = 0;
    
    data.sort((a, b) => {
        if(a[col - 1] === b[col - 1]) {
            return b[0] - a[0];
        } else {
            return a[col -1] - b[col - 1]
        }
    })
    
    for(var i = row_begin - 1; i < row_end; i++) {
        var modSum = data[i].reduce((cur, item) => {
            return cur + item % (i + 1);
        }, 0)
        answer = answer ^ modSum;
    }
    
    return answer;
}
